libp2p/src/main/kotlin/io/libp2p/mux/yamux/YamuxHandler.kt

```diff
@@ -75,6 +75,7 @@ open class YamuxHandler(
         when (msg.flags) {
             YamuxFlags.SYN -> {
                 // ACK the new stream
+                onStreamCreate(msg.id) // sometimes writes can happen before onRemoteCreated is called
                 onRemoteOpen(msg.id)
```

Comment:
That would result to `onStreamCreate()` is called twice. 
What's the intention of calling `onStreamCreate()` explicitly here?

Reply:
The send window needs to be initialized before onRemoteOpen, otherwise it errors and fails to do the first write.